src/main/java/tools/jackson/databind/deser/BeanDeserializerFactory.java

```
@@ -935,14 +936,15 @@ protected SettableBeanProperty constructSettableProperty(DeserializationContext
         // Does the Method specify the deserializer to use? If so, let's use it.
         TypeDeserializer typeDeser = (TypeDeserializer) type.getTypeHandler();
         SettableBeanProperty prop;
-        if (mutator instanceof AnnotatedMethod) {
-            prop = new MethodProperty(propDef, type, typeDeser,
-                    beanDesc.getClassAnnotations(), (AnnotatedMethod) mutator);
-        } else {
-            // 08-Sep-2016, tatu: wonder if we should verify it is `AnnotatedField` to be safe?
-            prop = new FieldProperty(propDef, type, typeDeser,
-                    beanDesc.getClassAnnotations(), (AnnotatedField) mutator);
+        // 06-04-2025, scs: we cannot always see members if e.g. they are in a different module that does not
```

Do we need to take a performance hit here? Why not let this fail if access is not setup? Null seems like a hack to me. The Java exceptions are fairly informative about the right add-opens are needed. @cowtowncoder wdyt? This looks like something we could do without. I only have a mobile to check this so the small screen view might not be the right way to check this PR.

---

Not sure what performance hit you are thinking -- this is not perf sensitive path (it's during deserializer construction).

I agree it seems odd to return `null` but I don't remember reasoning over straight failure.
@stevenschlansker is author here.
One thing to note tho is that `ClassUtil.checkAndFixAccess` just blindly overrides access if I recall and does not attempt to determine if call would likely fail -- it is just based on configuration settings. That could probably be improved, but we cannot test call here (we are not deserializing, we are building deserializer) to make sure.

---

From memory, the reason for this is because the access checks moved from "Method invoke time" to "MethodHandle create time" - this is one of the potential benefits of MH, lifting access / visibility checks from each-invocation to once-at-creation.

Some JDK classes in the modular world are not visible to user code - you're not allowed to crack open java.lang anymore, for example, without various `--add-opens` directives.

Jackson creates deserializers for types that are not used in some cases (I don't recall which) but basically this is skipping a failure in the case where we see some JDK type that we cannot crack open (Date IIRC?), but it does not matter, because the deserializer does not get used. So we skip an irrelevant failure.

I agree `null` is a bit weird. Improvements welcome, but the code is that way for a reason at least :)

---

Ok. @pjfanning if you wanted to try a PR with explicit fail, we could see if any of unit tests fail? (or just test locally).

@stevenschlansker Yeah I do not doubt there was (is) a reason (I vaguely recall discussions). But also do not remember context well enough to know the "right way" to go.